--- libxslt-1.1.28/libexslt/Makefile.orig	2016-02-11 14:23:51.000000000 -0500
+++ libxslt-1.1.28/libexslt/Makefile	2016-02-11 14:30:34.000000000 -0500
@@ -347,7 +347,7 @@
 	dynamic.c
 
 libexslt_la_LIBADD = $(top_builddir)/libxslt/libxslt.la $(EXTRA_LIBS) $(LIBGCRYPT_LIBS)
-libexslt_la_LDFLAGS = $(WIN32_EXTRA_LDFLAGS) -version-info $(LIBEXSLT_VERSION_INFO)
+libexslt_la_LDFLAGS = $(WIN32_EXTRA_LDFLAGS) -version-info $(LIBEXSLT_VERSION_INFO) -Wl,--image-base,0x6d780000
 man_MANS = libexslt.3
 EXTRA_DIST = $(man_MANS)
 all: all-am
