diff --git a/lib/rubygems/platform.rb b/lib/rubygems/platform.rb
index 48b7344aee..08728c209c 100644
--- a/lib/rubygems/platform.rb
+++ b/lib/rubygems/platform.rb
@@ -1,7 +1,5 @@
 # frozen_string_literal: true

-require_relative "deprecate"
-
 ##
 # Available list of platforms for targeting Gem installations.
 #
@@ -24,11 +22,6 @@ def self.match(platform)
     match_platforms?(platform, Gem.platforms)
   end

-  class << self
-    extend Gem::Deprecate
-    rubygems_deprecate :match, "Gem::Platform.match_spec? or match_gem?"
-  end
-
   def self.match_platforms?(platform, platforms)
     platform = Gem::Platform.new(platform) unless platform.is_a?(Gem::Platform)
     platforms.any? do |local_platform|
