diff --git a/metasploit-framework.gemspec b/metasploit-framework.gemspec
index ca49e3c..cd7b3a7 100644
--- a/metasploit-framework.gemspec
+++ b/metasploit-framework.gemspec
@@ -24,9 +24,6 @@ Gem::Specification.new do |spec|
   spec.homepage      = 'https://www.metasploit.com'
   spec.license       = 'BSD-3-clause'
 
-  spec.files         = `git ls-files`.split($/).reject { |file|
-    file =~ /^documentation|^data\/gui|^external/
-  }
   spec.bindir = '.'
   if ENV['CREATE_BINSTUBS']
     spec.executables   = [
