diff --git a/metasploit-framework.gemspec b/metasploit-framework.gemspec
index b55732712a..137e74da10 100644
--- a/metasploit-framework.gemspec
+++ b/metasploit-framework.gemspec
@@ -88,7 +88,7 @@ Gem::Specification.new do |spec|
   # Used by the Metasploit data model, etc.
   # bound to 0.20 for Activerecord 4.2.8 deprecation warnings:
   # https://github.com/ged/ruby-pg/commit/c90ac644e861857ae75638eb6954b1cb49617090
-  spec.add_runtime_dependency 'pg', '1.1.4'
+  spec.add_runtime_dependency 'pg', '~> 0.20'
   # Run initializers for metasploit-concern, metasploit-credential, metasploit_data_models Rails::Engines
   spec.add_runtime_dependency 'railties'
   # required for OS fingerprinting
