diff --git a/Gemfile.lock b/Gemfile.lock
index a05bbfbdc7e..1a3cff9acc7 100644
--- Gemfile.lock       2024-05-17 12:32:16
+++ Gemfile.lock       2024-05-17 12:32:59
@@ -32,7 +32,6 @@ PATH
       faraday (= 2.7.11)
       faraday-retry
       faye-websocket
-      ffi (< 1.17.0)
       fiddle
       filesize
       getoptlong
@@ -64,9 +63,7 @@ PATH
       openssl-ccm
       openvas-omp
       ostruct
-      packetfu
       patch_finder
-      pcaprub
       pdf-reader
       pg
       puma
@@ -380,14 +377,11 @@ GEM
     openssl-cmac (2.0.2)
     openvas-omp (0.0.4)
     ostruct (0.6.1)
-    packetfu (2.0.0)
-      pcaprub (~> 0.13.1)
     parallel (1.27.0)
     parser (3.3.8.0)
       ast (~> 2.4.1)
       racc
     patch_finder (1.0.2)
-    pcaprub (0.13.3)
     pdf-reader (2.14.1)
       Ascii85 (>= 1.0, < 3.0, != 2.0.0)
       afm (~> 0.2.1)
diff --git a/metasploit-framework.gemspec b/metasploit-framework.gemspec
index 8c620476dbc..959eb1284f8 100644
--- metasploit-framework.gemspec       2024-05-17 12:32:16
+++ metasploit-framework.gemspec       2024-05-17 12:32:59
@@ -87,9 +87,9 @@ Gem::Specification.new do |spec|
   # Needed by for XML parsing
   spec.add_runtime_dependency 'nokogiri'
   # Needed by db.rb and Msf::Exploit::Capture
-  spec.add_runtime_dependency 'packetfu'
+  #spec.add_runtime_dependency 'packetfu'
   # For sniffer and raw socket modules
-  spec.add_runtime_dependency 'pcaprub'
+  #spec.add_runtime_dependency 'pcaprub'
   # Used by the Metasploit data model, etc.
   # bound to 0.2x for Activerecord 4.2.8 deprecation warnings:
   # https://github.com/ged/ruby-pg/commit/c90ac644e861857ae75638eb6954b1cb49617090
@@ -154,7 +154,7 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency 'net-smtp'
   spec.add_runtime_dependency 'net-sftp'
   spec.add_runtime_dependency 'winrm'
-  spec.add_runtime_dependency 'ffi', '< 1.17.0'
+  #spec.add_runtime_dependency 'ffi', '< 1.17.0'
 
   #
   # REX Libraries
@@ -266,7 +266,6 @@ Gem::Specification.new do |spec|
     bigdecimal
     csv
     drb
-    fiddle
     getoptlong
     mutex_m
     ostruct
