--- template/prelude.c.tmpl.real	2018-01-25 12:12:58.636606731 +0000
+++ template/prelude.c.tmpl	2018-01-25 12:14:02.802380256 +0000
@@ -142,7 +142,7 @@
 static void
 prelude_eval(VALUE code, VALUE name, int line)
 {
-#ifdef __GNUC__
+#if defined __GNUC__ && __GNUC__ >= 5
 # pragma GCC diagnostic push
 # pragma GCC diagnostic error "-Wmissing-field-initializers"
 #endif
@@ -159,7 +159,7 @@
 	FALSE, /* unsigned int coverage_enabled; */
 	0, /* int debug_level; */
     };
-#ifdef __GNUC__
+#if defined __GNUC__ && __GNUC__ >= 5
 # pragma GCC diagnostic pop
 #endif
 
