--- a/load.c
+++ b/load.c
@@ -605,7 +605,7 @@ rb_load_internal0(rb_execution_context_t *ec, VALUE fname, int wrap)
 	    rb_parser_set_context(parser, NULL, FALSE);
 	    ast = (rb_ast_t *)rb_parser_load_file(parser, fname);
 	    iseq = rb_iseq_new_top(&ast->body, rb_fstring_lit("<top (required)>"),
-			    fname, rb_realpath_internal(Qnil, fname, 1), NULL);
+			    fname, fname, NULL);
 	    rb_ast_dispose(ast);
 	}
         rb_exec_event_hook_script_compiled(ec, iseq, Qnil);
