--- ruby-2.6.7-orig/hash.c	2021-04-05 04:48:34.000000000 -0700
+++ ruby-2.6.7/hash.c	2021-04-16 13:50:06.000000000 -0700
@@ -5658,7 +5658,8 @@
     long keylen, elen;
     const char *keyptr, *eptr;
     RSTRING_GETMEM(key, keyptr, keylen);
-    for (long i=0; i<RARRAY_LEN(keys); i++) {
+    long i;
+    for (i=0; i<RARRAY_LEN(keys); i++) {
         VALUE e = RARRAY_AREF(keys, i);
         RSTRING_GETMEM(e, eptr, elen);
         if (elen != keylen) continue;
