diff --git a/include/yaml.h b/include/yaml.h
index 5a04d36..70d839e 100644
--- a/include/yaml.h
+++ b/include/yaml.h
@@ -29,7 +29,7 @@ extern "C" {
 #ifdef _WIN32
 #   if defined(YAML_DECLARE_STATIC)
 #       define  YAML_DECLARE(type)  type
-#   elif defined(YAML_DECLARE_EXPORT)
+#   elif defined(YAML_DECLARE_EXPORT) || defined(DLL_EXPORT)
 #       define  YAML_DECLARE(type)  __declspec(dllexport) type
 #   else
 #       define  YAML_DECLARE(type)  __declspec(dllimport) type
diff --git a/src/Makefile b/src/Makefile
index b72224d..a413ce0 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -219,7 +219,7 @@ top_srcdir = ..
 AM_CPPFLAGS = -I$(top_srcdir)/include
 lib_LTLIBRARIES = libyaml.la
 libyaml_la_SOURCES = yaml_private.h api.c reader.c scanner.c parser.c loader.c writer.c emitter.c dumper.c
-libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE)
+libyaml_la_LDFLAGS = -release $(YAML_LT_RELEASE) -version-info $(YAML_LT_CURRENT):$(YAML_LT_REVISION):$(YAML_LT_AGE) -no-undefined
 all: all-am
 
 .SUFFIXES:
